### PR TITLE
fix broadcast for user protocol

### PR DESCRIPTION
**Pull request by pshemass**

### CODE REVIEW

File: examples/src/main/scala/zio/keeper/example/Example.scala [hunk 1]
```diff
@@ -60,37 +61,37 @@ object TestNode {
...
-  def discovery(others: Set[Int]): Managed[Exception, Layer[Nothing, Discovery]] =
-    ZManaged
-      .foreach(others) { port =>
-        InetAddress.localHost.flatMap(SocketAddress.inetSocketAddress(_, port)).toManaged_
-      }
-      .orDie
-      .map(addrs => Discovery.staticList(addrs.toSet))
+    (for {
+      _ <- sleep(5.seconds)
+      _ <- events[PingPong].foreach(event => putStrLn("membership event: " + event)).fork
```
Comment (mijicd) [inline examples/src/main/scala/zio/keeper/example/Example.scala:66]: I'd replace these with ZIO logging.

File: keeper/src/main/scala/zio/keeper/membership/swim/Broadcast.scala [hunk 2]
```diff
@@ -30,7 +36,10 @@ final class Broadcast(ref: TRef[TreeSet[Item]], sequenceId: TRef[Int], messageOv
           (toSend, toReschedule :+ item, size)
       }
 
-      val newValue  = TreeSet.empty[Item] ++ toSend.map(item => item.copy(resend = item.resend - 1)) ++ toReschedule
+      val newValue = TreeSet.empty[Item] ++
```
Comment (mijicd) [inline keeper/src/main/scala/zio/keeper/membership/swim/Broadcast.scala:39]: Is TreeSet.empty needed here?

File: keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala [hunk 3]
```diff
@@ -46,13 +46,13 @@ final class Nodes(
               .commit
               .tap(
                 _ => {
-                  val event = if (newState == NodeState.Healthy && prev == NodeState.Init) {
-                    Join(id)
+                  if (newState == NodeState.Healthy && prev == NodeState.Init) {
```
Comment (mijicd) [inline keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala:49]: This can be polished with ZIO.whenCaseM.

Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala:49]: I knew that you gonna say that :)

File: examples/src/main/scala/zio/keeper/example/Example.scala (hunk 1 above)
Comment (pshemass) [reply examples/src/main/scala/zio/keeper/example/Example.scala:66]: sure

File: keeper/src/main/scala/zio/keeper/membership/swim/Broadcast.scala (hunk 2 above)
Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/membership/swim/Broadcast.scala:39]: let me check

**Pull request merged**